beacon_node/lighthouse_network/src/peer_manager/mod.rs

```diff
@@ -819,26 +833,290 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
+                                        outbound_peers_pruned += 1;
+                                    } else {
+                                        // Restart the main loop with the outbound peer removed from
+                                        // the list. This will lower the peers per subnet count and
+                                        // potentially a new subnet may be chosen to remove peers. This
+                                        // can occur recursively until we have no peers left to choose
+                                        // from.
+                                        continue;
+                                    }
+                                }
+
+                                // Check the sync committee
+                                if let Some(subnets) = peer_to_sync_committee.get(candidate_peer) {
+                                    // The peer is subscribed to some long-lived sync-committees
+                                    // Of all the subnets this peer is subscribed too, the minimum
+                                    // peer count of all of them is min_subnet_count
+                                    if let Some(min_subnet_count) = subnets
+                                        .iter()
+                                        .map(|v| {
+                                            *sync_committee_peer_count.get(v).unwrap_or(&100u64)
+                                        })
+                                        .min()
+                                    {
+                                        // If the minimum count is our target or lower, we
+                                        // shouldn't remove this peer, because it drops us lower
+                                        // than our target
+                                        if min_subnet_count <= MIN_SYNC_COMMITTEE_PEERS {
+                                            // Do not drop this peer in this pruning interval
+                                            continue;
+                                        }
+                                    }
+                                }
+
+                                // This peer is suitable to be pruned
+                                removed_peer_index = Some(index);
+                                break;
+                            }
+
+                            // If we have successfully found a candidate peer to prune, prune it,
+                            // otherwise all peers on this subnet should not be removed due to our
+                            // outbound limit or min_subnet_count. In this case, we remove all
+                            // peers from the pruning logic and try another subnet.
+                            if let Some(index) = removed_peer_index {
+                                let (candidate_peer, _) = peers_on_subnet.remove(index);
+                                // Remove pruned peers from other subnet counts
```

Comment:
Should we also update the sync committee counts here?

If we don't I think we run the risk of pruning sync committee peers based on stale data E.g. if all sync subnets have 3 peers each and we prune a peer that is part of all of them, then we'll have 2 peers per sync subnet and shouldn't prune any more sync peers. However because we haven't updated the `sync_committee_peer_count` map it will continue returning 3 and we may end up pruning another peer which is connected to all the sync committee subnets, dropping us to 1 peer per sync subnet

I don't think we need to update `peer_to_sync_committee` however

Reply:
Yeah! Nice catch